src/web/src/app.tsx

```diff
@@ -0,0 +1,362 @@
+  if (data === undefined) return undefined;
+  return await processSeed(data, type);
+}
+
+export function FeatureItem(props: {
+  feature: FeatureModel;
+  className: string;
+}): JSX.Element {
+  return (
+    <a
+      className={props.className}
+      target="blank"
+      rel="noreferrer"
+      href={props.feature.link}
+    >
+      {props.feature.name}
+    </a>
+  );
+}
+
+export function FeatureList(props: {
+  title: string;
+  className: string;
+  features: FeatureModel[];
+}): JSX.Element {
+  if (props.features.length === 0) {
+    return <></>;
+  }
+  const features = props.features.map((f) => (
+    <li key={f.name}>
+      <FeatureItem feature={f} className={props.className} />
+    </li>
+  ));
+  return (
+    <ul className="study-meta">
+      <span>{props.title}: </span>
+      {features}
+    </ul>
+  );
+}
+
+export function ExperimentItem(props: { exp: ExperimentModel }): JSX.Element {
+  const paramsList = props.exp.parameters().map((p) => <li key={p}>{p}</li>);
+  const paramsDiv =
+    paramsList.length > 0 ? (
+      <ul className="study-meta">
+        <span>Parameters: </span>
+        {paramsList}
+      </ul>
+    ) : (
+      <></>
+    );
+  const classes =
+    'list-group-item ' +
+    (props.exp.isMajorGroup() ? 'major-exp-item' : 'exp-item');
+  return (
+    <li className={classes}>
+      {props.exp.name()} ({props.exp.weight()}%)
+      <FeatureList
+        title="Enabled features"
+        className="enabled-feature"
+        features={props.exp.enabledFeatures()}
+      />
+      <FeatureList
+        title="Disabled features"
+        className="disabled-feature"
+        features={props.exp.disabledFeatures()}
+      />
+      {paramsDiv}
+    </li>
+  );
+}
+
+export function PropertyList(props: {
+  caption: string;
+  list: string[] | null | undefined;
+}): JSX.Element {
+  if (props.list == null || props.list.length === 0) return <></>;
+  return (
+    <ul className="study-meta">
+      <span>{props.caption}: </span>
+      {props.list.map((c) => (
+        <li key={c}>{c}</li>
+      ))}
+    </ul>
+  );
+}
+
+export function IncludeExcludeList(props: {
+  caption: string;
+  include: string[] | null | undefined;
+  exclude: string[] | null | undefined;
+}): JSX.Element {
+  return (
+    <>
+      <PropertyList caption={props.caption} list={props.include} />
+      <PropertyList caption={'Exclude' + props.caption} list={props.exclude} />
+    </>
+  );
+}
+
+export function StudyItem(props: {
+  study: StudyModel;
+  filter: StudyFilter;
+}): JSX.Element {
+  const filter = props.study.filter();
+  return (
+    <div className="card mb-3">
+      <div className="card-header">
+        <a target="_blank" href={props.study.getConfigUrl()} rel="noreferrer">
+          {props.study.name()}
+        </a>
+      </div>
+      <div className="card-body">
+        <ul className="list-group list-group-flush">
+          {props.study.experiments(props.filter).map((e) => (
```

Comment:
This seems expensive. React functions are called anytime anything on the page changes at all. Perhaps we want to memoize it https://react.dev/reference/react/useMemo?
```tsx
const experiments = React.useMemo(() => props.study.experiments(props.filter), [props.study, props.filter])
```

Reply:
done.